src/cpp/client.cpp

```diff
@@ -1366,16 +1552,18 @@ std::vector<DataSet> Client::get_dataset_list_range(const std::string& list_name
 void Client::_set_prefixes_from_env()
 {
     // Establish set prefix
-    const char* keyout_p = std::getenv("SSKEYOUT");
-    if (keyout_p != NULL)
-        _put_key_prefix = keyout_p;
+    std::string put_key_prefix;
+    get_string_from_env(put_key_prefix, "SSKEYOUT", "");
+    if (put_key_prefix.length() > 0)
+        _put_key_prefix = put_key_prefix;
```

Comment:
Is there a c++ equivalent of `IConfiguration` that could be passed to the constructor? 

By passing configuration in, you can decouple where the config comes from (and build hierarchical config, use a config file, change env var names without churn, etc).

Reply:
Let's chat -- I'm not sure I understand what you're suggesting here